AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -553,8 +551,13 @@ - (void)_loadViewOrLayerIsLayerBacked:(BOOL)isLayerBacked
   if (isLayerBacked) {
     TIME_SCOPED(_debugTimeToCreateView);
     _layer = [self _layerToLoad];
-    // Surpress warning for Base SDK > 10.0
-    _layer.delegate = (id<CALayerDelegate>)self;
+    static int ASLayerDelegateAssociationKey;
+
+    // We cannot make self the layer's delegate directly, because the delegate is
+    // actually `assign` though the docs say weak.
+    ASWeakProxy *instance = [ASWeakProxy weakProxyWithTarget:self];
+    _layer.delegate = (id<CALayerDelegate>)instance;
+    objc_setAssociatedObject(_layer, &ASLayerDelegateAssociationKey, instance, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
```

Comment:
Yep, it's not free. Still, we need to land this and move on. The performance cost here is almost certainly not worth straining our already-strained engineering resources. Especially when there is incredibly low-hanging performance fruit waiting for us up ahead.

One of the important uses for the layer delegate is observing kCAOrderIn/kCAOrderOut. We cannot have the layer be its own delegate unless we drop support for that or drop support for layer-backed synchronous nodes. Which I would be OK with, although AVPlayerLayer nodes are nice to have.